lib/si-layer-cache/src/retry_queue.rs

```diff
@@ -0,0 +1,609 @@
+//! - `initial_backoff`: Starting delay (default: 100ms)
+//! - `max_backoff`: Maximum delay (default: 5s)
+//! - `backoff_multiplier`: Growth factor (default: 2.0)
+
+use std::{
+    collections::{
+        BTreeSet,
+        HashMap,
+    },
+    ffi::OsString,
+    path::PathBuf,
+    time::{
+        Duration,
+        Instant,
+    },
+};
+
+use tokio::sync::mpsc;
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    error::{
+        LayerDbError,
+        LayerDbResult,
+    },
+    event::LayeredEvent,
+};
+
+/// Configuration for retry queue behavior
+#[derive(Debug, Clone)]
+pub struct RetryQueueConfig {
+    /// Base directory for all retry queues
+    pub base_path: PathBuf,
+    /// Initial backoff duration (default: 100ms)
+    pub initial_backoff: Duration,
+    /// Maximum backoff duration (default: 5 seconds)
+    pub max_backoff: Duration,
+    /// Backoff multiplier (default: 2.0)
+    pub backoff_multiplier: f64,
+}
+
+impl Default for RetryQueueConfig {
+    fn default() -> Self {
+        Self {
+            base_path: PathBuf::from("retry_queues"),
+            initial_backoff: Duration::from_millis(100),
+            max_backoff: Duration::from_secs(5),
+            backoff_multiplier: 2.0,
+        }
+    }
+}
+
+/// Tracks backoff state for a single queue
+#[derive(Debug)]
+struct QueueState {
+    retry_dir: PathBuf,
+    current_backoff: Duration,
+    next_retry_time: Instant,
+    pending_files: BTreeSet<OsString>,
+}
+
+impl QueueState {
+    fn new(retry_dir: PathBuf, initial_backoff: Duration) -> Self {
+        Self {
+            retry_dir,
+            current_backoff: initial_backoff,
+            next_retry_time: Instant::now(),
+            pending_files: BTreeSet::new(),
+        }
+    }
+
+    /// Record a failed retry attempt and update backoff
+    fn record_failure(&mut self, max_backoff: Duration, multiplier: f64) {
+        self.current_backoff =
+            calculate_next_backoff(self.current_backoff, max_backoff, multiplier);
+        self.next_retry_time = Instant::now() + self.current_backoff;
+    }
+
+    /// Record a successful retry and reset backoff
+    fn record_success(&mut self, initial_backoff: Duration) {
+        self.current_backoff = initial_backoff;
+        self.next_retry_time = Instant::now();
+    }
+
+    /// Check if this queue is ready to retry
+    fn is_ready(&self) -> bool {
+        Instant::now() >= self.next_retry_time && !self.pending_files.is_empty()
+    }
+}
+
+/// Handle for tracking a retry attempt
+#[derive(Debug)]
+pub struct RetryHandle {
+    pub(crate) cache_name: String,
+    pub(crate) filename: OsString,
+}
+
+/// Calculate the next backoff duration using exponential backoff with a maximum cap
+fn calculate_next_backoff(current: Duration, max: Duration, multiplier: f64) -> Duration {
+    std::cmp::min(
+        Duration::from_secs_f64(current.as_secs_f64() * multiplier),
+        max,
+    )
+}
```

Comment:
Not a deal breaker, but this https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/ has me convinced backoff should always be "jittered" (e.g: https://github.com/systeminit/si/blob/f2d11382d0d9fa73b9efbe75d9e0c5e600ae8e81/lib/dal/src/job/consumer.rs#L293)